torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -19,11 +21,37 @@
 #include <torch/csrc/jit/serialization/pickler.h>
 #include <torch/csrc/jit/serialization/unpickler.h>
 
-#include <fmt/format.h>
-
 namespace torch {
 namespace distributed {
 namespace rpc {
+namespace {
+void processRemoteProfiledEvents(
+    autograd::RpcWithProfilingResp& rpcWithProfilingResp) {
+  // Check if the profiler is enabled
+  auto enabled = torch::autograd::profiler::profilerEnabled();
+  TORCH_CHECK(
+      enabled,
+      "Profiler was expected to be enabled. This can happen in callback "
+      " continutations that run in different threads, and the TLS of the "
+      " profiler was not propagated.");
+  std::vector<torch::autograd::profiler::Event> events =
+      rpcWithProfilingResp.getProfiledEvents();
+  auto profilingId = rpcWithProfilingResp.getProfilingId();
```

Comment:
we can let `getProfilingId()` to return a `const&` and modify `retrieveRPCProfilingKey` and `eraseKey` accordingly to avoid the copy here?